src/elements/common/sub-header/ViewModeChangeButton.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import React from 'react';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import classNames from 'classnames';
+
+import Button from '../../../components/button';
+import IconGridViewInverted from '../../../icons/general/IconGridViewInverted';
+import IconListView from '../../../icons/general/IconListView';
+import messages from '../messages';
+import Tooltip from '../Tooltip';
+import type { ViewMode } from '../flowTypes';
+import { VIEW_MODE_GRID, VIEW_MODE_LIST } from '../../../constants';
+import './ViewModeChangeButton.scss';
+
+type Props = {
+    className?: string,
+    onViewModeChange: Function,
+    viewMode: ViewMode,
+} & InjectIntlProvidedProps;
+
+const ViewModeChangeButton = ({ className = '', onViewModeChange, intl, viewMode, ...rest }: Props) => {
+    const isGridView = viewMode === VIEW_MODE_GRID;
+
+    const onClick = () => {
+        onViewModeChange(isGridView ? VIEW_MODE_LIST : VIEW_MODE_GRID);
+    };
+
+    return (
+        <Tooltip
+            text={
+                isGridView ? <FormattedMessage {...messages.listView} /> : <FormattedMessage {...messages.gridView} />
+            }
+        >
+            <Button
+                className={classNames('bdl-ViewModeChangeButton', className)}
+                aria-label={isGridView ? intl.formatMessage(messages.listView) : intl.formatMessage(messages.gridView)}
```

Comment:
It looks like `Tooltip` adds `aria-describedby` to its child component, so this is probably not needed.

Reply:
I looked a little further into this, and `Tooltip` only adds `aria-describedby` to its child component when the tooltip is being shown.  (It also only adds a generic label like `aria-describedby="tooltip206"`) . At least for Cypress testing, I think we might need a specific aria-label.

Reply:
The tooltip should be shown on focus and hover, which is also when it'll be read for screen readers. The `aria-describedby` attribute is meant to associate the element with another element using the latter's unique `id`, so it seems like it's working as we intend.